model/common_structures.smithy

```diff
@@ -59,3 +59,34 @@ structure Hits{
     _source: Document,
     fields: Document
 }
+
+structure IndexPermission {
+    index_patterns: IndexPatterns
+    fls: Fls
+    masked_fields: MaskedFields
+    allowed_actions: AllowedActions
+}
+
+structure Role{
+    reserved: Boolean
+    hidden: Boolean
+    description: String
+    cluster_permission: ClusterPermission
+    index_permission: IndexPermission
+    tenant_permissions: TenantPermission
+    static: Boolean
+}
+
+list RolesList{
+    member: Role
+}
+
+structure Response{
```

Comment:
Some of the common structures are defined in mutliple PRs, I suggest we wait for one PR https://github.com/opensearch-project/opensearch-api-specification/pull/108 to be merged and then rebase all the others to use that.